### PR TITLE
Upgrade delta-sharing-spark to scala2.13

### PR DESCRIPTION
**Pull request by linzhou-db**
Upgrade delta-sharing-spark to scala2.13, original PR: https://github.com/delta-io/delta-sharing/pull/293

### CODE REVIEW

**Review by charlenelyu-db — APPROVED** [review]

File: spark/src/test/scala/io/delta/sharing/spark/DeltaSharingFileProfileProviderSuite.scala [hunk 1]
```diff
@@ -70,8 +70,8 @@ class DeltaSharingFileProfileProviderSuite extends SparkFunSuite {
     )
   }
 
-  test("version is missing") {
-    val e = intercept[IllegalArgumentException] {
+  test("shareCredentialsVersion is missing") {
+    val e = intercept[Exception] {
```
Comment (chakankardb) [inline spark/src/test/scala/io/delta/sharing/spark/DeltaSharingFileProfileProviderSuite.scala:74]: Question: why was this needed? Did the type of error change?

File: spark/src/test/scala/io/delta/sharing/spark/DeltaSharingSourceSuite.scala [hunk 2]
```diff
@@ -452,29 +452,36 @@ class DeltaSharingSourceSuite extends QueryTest
...
-          (f.getCanonicalPath.endsWith("2") || f.getCanonicalPath.endsWith("3"))) {
+          (f.getCanonicalPath.endsWith("2") || f.getCanonicalPath.endsWith("3") ||
+            f.getCanonicalPath.endsWith("2.crc") || f.getCanonicalPath.endsWith("3.crc") ||
+            f.getCanonicalPath.endsWith("metadata.crc"))) {
           f.delete()
         }
+        // SparkStructuredStreaming requires query id match to reuse a checkpoint.
+        if (f.getCanonicalPath.endsWith("metadata")) {
+          FileUtils.writeStringToFile(f, s"""{"id":"${newQuery.id}"}""")
+        }
```
Comment (chakankardb) [inline spark/src/test/scala/io/delta/sharing/spark/DeltaSharingSourceSuite.scala:473]: Can you please add some comments on what does "*metadata" file contain, and why are we overwriting the existing file?

File: spark/src/test/scala/io/delta/sharing/spark/DeltaSharingFileProfileProviderSuite.scala (hunk 1 above)
Comment (linzhou-db) [reply spark/src/test/scala/io/delta/sharing/spark/DeltaSharingFileProfileProviderSuite.scala]: The test passed when I changed it back to `IllegalArgumentException` in local, let's see if the PR checks will pass.

File: spark/src/test/scala/io/delta/sharing/spark/DeltaSharingSourceSuite.scala (hunk 2 above)
Comment (linzhou-db) [reply spark/src/test/scala/io/delta/sharing/spark/DeltaSharingSourceSuite.scala:475]: added a comment.

**Review by chakankardb — APPROVED** [review]
